### DISCUSSION TITLE
nested loops interruption with break X, continue X

### DISCUSSION THREAD

**Discussion by VardanKosakyan** [dotnet/csharplang#5521] [Language Ideas] [answered]
**Proposal**

 The problem of interrupting nested loops, I believe, has been discussed many times. The solution I want to propose is based on specifying the number of interrupted loops after the word "break", as in some other languages. With the difference that the nesting level is specified in Roman numerals.  
break II
…
break IV
...
break all 
Same with continue keyword. 
Roman numerals look more elegant and don't look like hardcoded code. 

**Example**
```c
foreach (var text in texts)
{
    foreach (var sentence in text.Split('.'))
    {
        foreach (var word in sentence.Split(' '))
        {
            if (word == "hello")
            {
                //some logic
                break II;
            }
            else if (word == "world")
            {
                //some logic
                break all;
            }
        }
    }
}

```
in this example
search in text is interrupted when “hello” found
search in all texts is interrupted when “world” found

**Comment by huoyaoyuan** [comment]
See #1597

**Reply by VardanKosakyan** [reply]
in this proposal labels are not used,  instead, the nesting level is indicated.  In my opinion coming up with names for labels is an additional waste of time for developers

**Reply by siegfriedpammer** [reply]
Counting the nesting level and having to deal with unintuitive roman numerals is more error-prone than having to think about one's code and coming up with sensible names.

**Reply by VardanKosakyan** [reply]
we only  need  to know outer levels from break point,  in most cases it will not be more than 3,  besides, "break all" can be used

**Comment by CyrusNajmabadi** [comment]
I would not be opposed to the idea of targeted breaks. However, I can say with confidence that the ldm would never approve using roman numerals here. That's a complete non starter.

**Comment by koszeggy** [comment, marked as answer]
I actually prefer this idea better than [labeled breaks/continues](https://github.com/dotnet/csharplang/issues/1597) (which I consider only as an unnecessary workaround for gotophobia). But it definitely should use normal integer literals instead of roman numbers.

**Comment by CyrusNajmabadi** [comment]
Proposal started here: https://github.com/dotnet/csharplang/blob/main/proposals/labeled-break-continue.md
Championed Issue here: https://github.com/dotnet/csharplang/issues/9875
Discussion here: https://github.com/dotnet/csharplang/discussions/9876

Closing this out to move discussion there.
